Add shader time scaling (3.2)

RandomShaper opened the pull request:
Shaders' `TIME` will be affected by the new shader time scale, that is set via the new `VisualServer::set_time_scale()` method.

~~Another builtin `TIME_UNSCALED` is added, that contains the accumulated time unaffected by the scale.~~
**UPDATE:** This part has been removed at @reduz's request, the rationale being that the it's such a big change for a small benefit: there are other means (like uniforms) to alter the passage of time for the few cases of shaders that need not to be affected by the scale. And I agree.

Closes #27127. (See it for discussion.)

**No version of this branch for 4.0 is needed, since it introduces global shader variables, with which one can easily implement their own version of this.**

reduz commented:
Please dont do TIME_UNSCALED, if user really wants this, let him/her pass a uniform with whathever is desired.

akien-mga commented:
I had misunderstood reduz's comment so here's a clarification:
```
23:30 <Akien> reduz: ah I see, so you mean TIME can be affected by the scale, and for the few shaders that need it unscaled (like pause menu effects) users can use uniforms?
23:30 <reduz> yeah
```

RandomShaper committed:
Add shader time scaling

Shaders' `TIME` will be affected by the new shader time scale, that is set via the also new `VisualServer::set_time_scale()`.

akien-mga commented:
Thanks!
